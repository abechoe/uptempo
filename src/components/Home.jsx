import Radium from 'radium';
import React from 'react';
import { Link } from 'react-router-dom';

const StyledLink = Radium(Link);

class Home extends React.Component {
  render() {
    return (
      <ul style={styles.base}>
        <li style={styles.li}>
          <StyledLink to="/new" style={styles.a}>Create an Event</StyledLink>
        </li>
      </ul>
    );
  }
}

const styles = {
  base: {
    font: '20px Helvetica, sans-serif',
  },

  li: {
    listStyle: 'none',
  },

  a: {
    textDecoration: 'none',
    ':hover': {
      textDecoration: 'underline',
    },
  },
}

export default Radium(Home);