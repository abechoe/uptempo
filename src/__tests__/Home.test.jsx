import React from 'react';
import { shallow } from 'enzyme';

import Home from 'components/Home';

describe('Home component tests', () => {
  it('renders the Home component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});