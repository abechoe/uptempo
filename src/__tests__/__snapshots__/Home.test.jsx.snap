// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests renders the Home component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <li
        data-radium={true}
        style={
          Object {
            "listStyle": "none",
          }
        }
      >
        <Link
          replace={false}
          style={
            Object {
              ":hover": Object {
                "textDecoration": "underline",
              },
              "textDecoration": "none",
            }
          }
          to="/new"
        >
          Create an Event
        </Link>
      </li>,
      "data-radium": true,
      "style": Object {
        "font": "20px Helvetica, sans-serif",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          style={
            Object {
              ":hover": Object {
                "textDecoration": "underline",
              },
              "textDecoration": "none",
            }
          }
          to="/new"
        >
          Create an Event
        </Link>,
        "data-radium": true,
        "style": Object {
          "listStyle": "none",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Create an Event",
          "replace": false,
          "style": Object {
            ":hover": Object {
              "textDecoration": "underline",
            },
            "textDecoration": "none",
          },
          "to": "/new",
        },
        "ref": null,
        "rendered": "Create an Event",
        "type": [Function],
      },
      "type": "li",
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li
          data-radium={true}
          style={
            Object {
              "listStyle": "none",
            }
          }
        >
          <Link
            replace={false}
            style={
              Object {
                ":hover": Object {
                  "textDecoration": "underline",
                },
                "textDecoration": "none",
              }
            }
            to="/new"
          >
            Create an Event
          </Link>
        </li>,
        "data-radium": true,
        "style": Object {
          "font": "20px Helvetica, sans-serif",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            style={
              Object {
                ":hover": Object {
                  "textDecoration": "underline",
                },
                "textDecoration": "none",
              }
            }
            to="/new"
          >
            Create an Event
          </Link>,
          "data-radium": true,
          "style": Object {
            "listStyle": "none",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Create an Event",
            "replace": false,
            "style": Object {
              ":hover": Object {
                "textDecoration": "underline",
              },
              "textDecoration": "none",
            },
            "to": "/new",
          },
          "ref": null,
          "rendered": "Create an Event",
          "type": [Function],
        },
        "type": "li",
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
